### PR TITLE
Scenario detail view

### PR DESCRIPTION
**Pull request by jeannie3**
Current changes include: 
- Moved ScenarioList to components folder
- Change saving from cookies to Local Storage (it was simpler with local storage)
- Added routing for ScenarioList
- Converted ScenarioDetailView from dialog to a regular component and page
- Connected ScenarioList and ScenarioDetailView
- Add scenario leads ScenarioDetailView (only when you save does it add to the scenario to the ScenarioList)

### CODE REVIEW

**Comment by sgundlapalli11** [comment]
1. Why is there no point in having routes for the `scenario-detail-dialog`? Think it would be helpful to still have `scenarios/:id`; the use case would just be to allow the user to press back in the browser from the dialog view 
2. Is it possible for the scenario-list-view component have a `/scenarios` route? 
3. If you update any of the scenario details and proceed to refresh the page, the updated information is stored in local storage but does not reflect in the `scenario-list-view` but will show up when you open the `scenario-detail-dialog`. Don't think this is a big deal and will be fixed when we incorporate database. Can send some screen recording replicating this if needed 
4. If you enter a scenario title that is too long, the scenarios in `scenario-list-view` overlap. 

Past that, everything looks great! Good addition of the wildcard route 😃

**Comment by jeannie3** [comment]
> 1. Why is there no point in having routes for the `scenario-detail-dialog`? Think it would be helpful to still have `scenarios/:id`; the use case would just be to allow the user to press back in the browser from the dialog view
> 2. Is it possible for the scenario-list-view component have a `/scenarios` route?
> 3. If you update any of the scenario details and proceed to refresh the page, the updated information is stored in local storage but does not reflect in the `scenario-list-view` but will show up when you open the `scenario-detail-dialog`. Don't think this is a big deal and will be fixed when we incorporate database. Can send some screen recording replicating this if needed
> 4. If you enter a scenario title that is too long, the scenarios in `scenario-list-view` overlap.
> 
> Past that, everything looks great! Good addition of the wildcard route 😃

1. I actually forgot about that use case, but if we do decide to use the back button, I think dialog might not be the best way to display the ScenarioDetailsView, because there will be the SceneDetailView, which will make it complicated with dialog, so I will convert the ScenarioDetailsView to a regular component/page to add routing nicely.
2. Yea it is possible. The only thing is that since we only have scenarios, it should fine to leave it just /.
3. Yea this is a problem I know about. It should be fixed with database, so I didn't try to go too far into fixing it and just left it as it was for now since we are going to be adding the database soon.
4. Thanks for noticing that. I'll fix it with some css

**Comment by hdang32** [comment]
@sgundlapalli11 
2) Is there anything else that we have beside our scenarios? For now we can eliminate "/scenarios" for the sake of developing it and if there is something else coming up, adding /scenarios as a prefix would be trivial. 
4) For the purpose of this sprint, I would suggest us **not** to worry much about styling.  Styling is tricky and it could consume a chunk of your time. I hardcoded the width of the scenario button on purpose. We haven't decided on the permitted length of a title nor anything else related to the datatype + size of the input data. We can always go back to perfect our components once we truly understand the client's requirement.

File: src/app/mock/mock-scenarios.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export const SCENARIOS: Scenario[] = [
```
Comment (hdang32) [inline src/app/mock/mock-scenarios.ts]: We don't need this guy anymore right?

**Comment by sgundlapalli11** [comment]
Good to merge 👍

**Pull request merged**